src/lib/viewers/controls/page/PageControlsForm.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import { decodeKeydown } from '../../../util';
+import './PageControlsForm.scss';
+
+export type Props = {
+    onPageSubmit: (newPageNumber: number) => void;
+    pageCount: number;
+    pageNumber: number;
+};
+
+export const ENTER = 'Enter';
+export const ESCAPE = 'Escape';
+
+export default function PageControlsForm({ onPageSubmit, pageNumber, pageCount }: Props): JSX.Element {
+    const [isInputShown, setIsInputShown] = React.useState(false);
+    const [inputValue, setInputValue] = React.useState(pageNumber);
+    const isButtonFocus = React.useRef(false);
+    const buttonElRef = React.useRef<HTMLButtonElement>(null);
+    const inputElRef = React.useRef<HTMLInputElement>(null);
+
+    const setPage = (allowRetry = false): void => {
+        if (!Number.isNaN(inputValue) && inputValue >= 1 && inputValue <= pageCount && inputValue !== pageNumber) {
+            onPageSubmit(inputValue);
+        } else {
+            setInputValue(pageNumber); // Reset the invalid input value to the current page
+
+            if (allowRetry) {
+                isButtonFocus.current = true;
+            }
+        }
+
+        setIsInputShown(false);
+    };
+
+    const handleNumInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        setInputValue(parseInt(event.target.value, 10));
+    };
+
+    const handleNumInputBlur = (): void => {
+        setPage();
+    };
+
+    const handleNumInputKeyDown = (event: React.KeyboardEvent<HTMLInputElement>): void => {
+        const key = decodeKeydown(event);
+
+        switch (key) {
+            case ENTER:
+                event.stopPropagation();
+                event.preventDefault();
+
+                setPage(true);
+                break;
+            case ESCAPE:
+                event.stopPropagation();
+                event.preventDefault();
+
+                setIsInputShown(false);
+                isButtonFocus.current = true;
+                break;
+            default:
+                break;
+        }
+    };
+
+    React.useEffect(() => {
+        setInputValue(pageNumber); // Keep internal state in sync with prop as it changes
+    }, [pageNumber]);
+
+    React.useLayoutEffect(() => {
+        if (buttonElRef.current && isButtonFocus) {
+            buttonElRef.current.focus();
+            isButtonFocus.current = false;
+        }
+    }, [isButtonFocus]);
+
+    React.useLayoutEffect(() => {
+        if (inputElRef.current && isInputShown) {
+            inputElRef.current.select();
+        }
+    }, [isInputShown]);
+
+    return (
+        <div className="bp-PageControlsForm">
+            {isInputShown ? (
+                <input
+                    ref={inputElRef}
+                    className="bp-PageControlsForm-input"
+                    data-testid="bp-PageControlsForm-input"
+                    disabled={pageCount <= 1}
```

Comment:
Is this needed? It's not present in the existing controls.

Reply:
I thought Conrad brought up a good point before about just using <= instead of ===.  I can revert it since we do have some checks in both the component and the viewer itself.